pallets/scheduler/src/lib.rs

```diff
@@ -1496,136 +1273,111 @@ impl<T: Config> Pallet<T> {
     }
 }
 
-#[allow(deprecated)]
-impl<T: Config> schedule::v2::Anon<BlockNumberFor<T>, <T as Config>::RuntimeCall, T::PalletsOrigin>
+impl<T: Config> schedule::v3::Anon<BlockNumberFor<T>, <T as Config>::RuntimeCall, T::PalletsOrigin>
     for Pallet<T>
 {
-    type Address = TaskAddress<BlockNumberFor<T>>;
-    type Hash = T::Hash;
+    type Address = TaskAddressOf<T>;
+    type Hasher = T::Hashing;
 
     fn schedule(
-        when: DispatchTime<BlockNumberFor<T>>,
+        when: DispatchBlock<BlockNumberFor<T>>,
         maybe_periodic: Option<schedule::Period<BlockNumberFor<T>>>,
```

Comment:
Here maybe_periodic is only a blocknumber, no moment...

is this because v3 is a legacy method?

Should we just remove legacy methods? Seems like we don't need old versions around.